inc/namespace.php

```diff
@@ -590,3 +591,51 @@ function get_report_only_content_security_policies() : array {
 	 */
 	return apply_filters( 'altis.security.browser.report_only_content_security_policies', $policies );
 }
+
+/**
+ * Get allowed CORS origins
+ * 
+ * @return array
+ */
+function get_allowed_cors_origins() : array {
+	$origins = [];
+
+	/**
+	 * Filter the allowed CORS origins.
+	 *
+	 * @param array $origins List of allowed origins.
+	 */
+	return apply_filters( 'altis.security.browser.rest_allow_origins', $origins );
+}
+
+/**
+ * Restrict CORS origin
+ */
+function restrict_cors_origin( $value ) {
+	$rest_allow_origin = apply_filters( 'altis.security.browser.rest_allow_origin', false );
+	// Remove CORS header
+	if ( ! $rest_allow_origin ) {
+		// Don't send CORS header if origin is explicitly not allowed
+		remove_filter( 'rest_pre_serve_request','rest_allow_origin' );
```

Comment:
I think we should have this trigger an error as well to explicitly fail the request; per [the spec](https://fetch.spec.whatwg.org/#http-responses), this should be a 403.

If we switch this function to the `rest_pre_dispatch` hook, we can return a `WP_Error` directly.

Reply:
Updated to use `rest_pre_dispatch` and throw `WP_Error` with status of 403